sympy/stats/crv_types.py

```diff
@@ -2425,6 +2425,12 @@ def Normal(name, mean, std):
         from sympy.stats.joint_rv_types import MultivariateNormalDistribution
         return multivariate_rv(
             MultivariateNormalDistribution, name, mean, std)
+    if isinstance(mean, RandomSymbol) and\
+        isinstance(mean.pspace.distribution, NormalDistribution) and\
```

Comment:
please put a comment explaining why you are doing this. People who read the code will likely not understand that it's handling the compound distribution, and will get confused by this code.

Reply:
I think it would be even better to define a new method to perform the transformations, so that the code remains separate. Maybe we could introduce a new class to handle compounded random symbols, which can then be solved in order to simplify them in such cases. @czgdp1807 what do you think?

Reply:
> I think it would be even better to define a new method to perform the transformations, so that the code remains separate.

Decoupling is a nice idea. 

> Maybe we could introduce a new class to handle compounded random symbols, which can then be solved in order to simplify them in such cases. 

Yes, this should be done. Currently I believe `JointPSpace` is being used for transferring operations to `MarginalDistribution` which leads to confusion while debugging. This I figured out while bug fixing, I don't know about the correctness of my conclusion. 
A structure should be formed in such a way that it is capable of using direct results where ever utmost necessary and also capable of using other `core` and `integrals` modules. Symbolic parameters should be taken care of from the start. Symbolic results should be given preference.
Something like, `CompoundPSpace` which contains mechanics for computations and some specific distributions for which the closed form exists already can be implemented. I think @ritesh99rakesh has mentioned about it in his proposal. 
If possible cases like, in [this paper](https://onlinelibrary.wiley.com/doi/pdf/10.1002/9780470012505.tac046) should be implemented, though this can be given a lower priority.
Thanks. Please suggest improvements on my idea.

Reply:
The `CompoundPSpace` idea looks nice. So just declare a `RandomSymbol` object with `CompoundPSpace` as its probability space. After this, add a function trying to perform simplifications of the compound probability space (i.e. recognize known combinations of random parameters leading to known probability spaces).

Reply:
Yes, that would be compatible with the current implementation in `sympy.stats`. I would say the current design of `sympy.stats` is good though there are some flaws here and there. I think we can discuss on that some time but the general design is nice like, a class for Probability Space which contains the mechanics and some specific distributions/processes which interact with the outer world via that PSpace class. @ritesh99rakesh are you in agreement with this design? If yes, then the finer details of implementations can be discussed once you make changes to this PR.

Reply:
Yes, I agree with the design. In fact, I proposed this idea with more details in my proposal and am currently working on the same. But I thought since we could directly identify the underlying distribution for certain compound distributions, it would be useful to directly call those. Nevertheless, after reading the above comment (https://github.com/sympy/sympy/pull/17036#discussion_r294272432) I feel that we can put the entire code as separate. I will update this PR accordingly.